bigquery_etl/cli/stage.py

```diff
@@ -260,13 +261,57 @@ def _udf_dependencies(artifact_files):
 def _view_dependencies(artifact_files, sql_dir):
     """Determine view dependencies."""
     view_dependencies = set()
-    view_dependency_files = [file for file in artifact_files if file.name == VIEW_FILE]
+    view_dependency_files = [
+        file for file in artifact_files if file.name in {VIEW_FILE, QUERY_FILE}
+    ]
     for dep_file in view_dependency_files:
         # all references views and tables need to be deployed in the same stage project
         if dep_file not in artifact_files:
             view_dependencies.add(dep_file)
 
-        if dep_file.name == VIEW_FILE:
+        if dep_file.name == QUERY_FILE:
+            # check query files for view dependencies
+            # views need to be deployed to stage as they might reference other changed artifacts upstream
+            # without getting changed themselves
+            dependencies = extract_table_references(
+                render(dep_file.name, dep_file.parent)
+            )
+            for dependency in dependencies:
+                dependency_components = dependency.split(".")
+                dependency_project_id, dependency_dataset_id, dependency_table_id = (
+                    None,
+                    None,
+                    None,
+                )
+
+                match len(dependency_components):
+                    case 3:
+                        dependency_project_id,
+                        dependency_dataset_id,
+                        dependency_table_id = dependency_components
```

Comment:
I was surprised this is valid Python syntax.  I know @kik-kik suggested not using list/tuple syntax for one of the unpacking assignment statements, but I think that was intended just for that case where it all fit on one line.  With Black auto-formatting this so poorly (IMO) I think using list/tuple syntax in this case would be more readable.